test/core/unreached-invalid.wast

```diff
@@ -249,6 +249,42 @@
   ))
   "type mismatch"
 )
+(assert_invalid
+  (module (func $type-unary-num-vs-void-in-block-after-unreachable
+    (unreachable) (block (drop (i32.eq (nop))))
+  ))
+  "type mismatch"
+)
+(assert_invalid
+  (module (func $type-unary-num-vs-void-in-loop-after-unreachable
+    (unreachable) (loop (drop (i32.eq (nop))))
+  ))
+  "type mismatch"
+)
+(assert_invalid
+  (module (func $type-unary-num-vs-void-in-i32-loop-after-unreachable
+    (unreachable) (loop (result i32) (i32.eq (nop)))
+  ))
+  "type mismatch"
+)
+(assert_invalid
+  (module (func $type-unary-num-vs-void-in-if-after-unreachable
+    (unreachable) (if (i32.const 0) (then (i32.eqz (nop))))
```

Comment:
You may want to add a `drop` around the `i32.eqz`, so that it fires for the right reason.

Reply:
Done